src/main/java/org/apache/commons/compress/archivers/zip/ZipFile.java

```diff
@@ -210,7 +222,21 @@ public ZipFile(final String name, final String encoding) throws IOException {
      * @throws IOException if an error occurs while reading the file.
      */
     public ZipFile(final File f, final String encoding) throws IOException {
-        this(f, encoding, true);
+        this(f.toPath(), encoding, true);
+    }
+
+    /**
+     * Opens the given path for reading, assuming the specified
+     * encoding for file names and scanning for unicode extra fields.
+     *
+     * @param path path to the archive.
+     * @param encoding the encoding to use for file names, use null
+     * for the platform's default encoding
```

Comment:
No need for extra whitespace.

Reply:
This is a copy paste from the existing javadocs of methods from within the class, if you want I can correct them all for white-spaces.